tools/typst-preview-frontend/src/ws.ts

```diff
@@ -393,30 +394,40 @@ export async function wsMain({ url, previewMode, isContentPreview }: WsArgs) {
         }));
 };
 
-function ensureInvertColors(root: HTMLElement | null, strategy: string) {
+/** The strategy to set invert colors, see editors/vscode/package.json for enum descriptions */
+const INVERT_COLORS_STRATEGY = ['never', 'auto', 'always'] as const;
+/** The value of strategy constant */
+type StrategyKey = (typeof INVERT_COLORS_STRATEGY)[number];
+/** The map of target */
+type StrategyMap = Partial<Record<'rest' | 'image', StrategyKey>>;
+
+function ensureInvertColors(root: HTMLElement | null, strategy: StrategyKey | StrategyMap) {
     if (!root) {
         return;
     }
 
-    let needInvertColor = false;
-    switch (strategy) {
-        case "never":
-        default:
-            break;
-        case "auto":
-            needInvertColor = determineInvertColor();
-            break;
-        case "always":
-            needInvertColor = true;
-            break;
+    // Uniforms type of strategy to `TargetMap`
+    if (typeof strategy === 'string') {
+        strategy = { rest: strategy };
     }
 
-    if (needInvertColor) {
-        root.classList.add("invert-colors");
-    } else {
-        root.classList.remove("invert-colors");
+    let autoDecision: { value: boolean } | undefined = undefined;
+    /** 
+     * Handles invert colors mode based on a string enumerated strategy.  
+     * @param strategy - The strategy set by user.
+     * @returns needInvertColor - Use or not use invert color.
+     */
+    const decide = (strategy: StrategyKey) => {
+        switch (strategy) {
+            case 'never': default: return false;
+            case 'auto': return (autoDecision ||= { value: determineInvertColor() }).value;
+            case 'always': return true;
+        }
     }
 
+    root.classList.toggle("invert-colors", decide(strategy?.rest || 'never'));
+    root.classList.toggle("normal-image", !decide(strategy?.image || strategy?.rest || 'never'));
```

Comment:
There are some subtle changes here.
- if the strategy of `image` is not specified, it inherits value from "rest" selector.